packages/credit-protocol/index.ts

```diff
@@ -418,15 +418,31 @@ export default class CreditProtocol {
     return config.gasPrice
   }
 
-  async getTxCost(currency: string, gasNeeded: number) {
+  async getTransactionCosts(currency: string, gasNeeded: number) {
     try {
       const gasPrice = await this.getGasPrice()
       const rate = await this.getEthExchange(currency)
+      const weiCost = gasPrice * gasNeeded
+      const ethCost = weiCost / WEI_PER_ETH
+      const currencyCost = Math.max( 0.01, ethCost * Number(rate) )
+
+      return {
+        ethCost,
+        ethCostFormatted: `${ethCost}`.slice(0,6),
+        currencyCost,
+        currencyCostFormatted: `${currencySymbols(currency)}${formatSettlementCurrencyAmount(String(currencyCost), false)}`,
+        weiCost
+      }
+    } catch (e) {
+    }
```

Comment:
Buried error here. Avoid buring errors like this where possible, as it makes the function significantly more difficult to debug; and if doing so as a last resort, include a comment describing why the error is being silently buried.

Reply:
In this case, what error, exactly, are we expecting to see? What would cause it to occur?

Reply:
This code was already here and is being left alone for this PR.

Reply:
This `catch` would catch either errors in the ethExchange or gasPrice calls, in other instances we add a `console.log` for the error, which might be good in this case.

Reply:
added a log message for now